docs: add modifyEnvironmentConfig hook (#2756)
web-infra-dev/rsbuild: website/docs/en/plugins/dev/hooks.mdx

});
```

### modifyEnvironmentConfig

> Modify the environment config passed to the Rsbuild, you can directly modify the config object, or return a new object to replace the previous object.

Reviewer: ```suggestion

Modify the Rsbuild configuration of a specific environment.

In the callback function, the config object in the parameters has already been merged with the common Rsbuild configuration. You can directly modify this config object, or you can return a new object to replace it.
```

- **Type:**

```ts
type ModifyEnvironmentConfigUtils = {

>   /** Current Environment name */

Reviewer: environment do not need to be uppercase

```suggestion
  /** Current environment name */
```

  name: string;
  mergeEnvironmentConfig: (
    ...configs: EnvironmentConfig[]
  ) => EnvironmentConfig;
};

function ModifyEnvironmentConfig(
  callback: (
    config: EnvironmentConfig,
    utils: ModifyEnvironmentConfigUtils,
  ) => MaybePromise<EnvironmentConfig | void>,
): void;
```

> - **Example:** Set a default value for a specific Environment configuration item：

Reviewer: ```suggestion
- **Example:** Set a default value for the Rsbuild config of a specified environment:
```
